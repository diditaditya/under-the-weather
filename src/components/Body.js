import React from 'react';
import {  View, Text, ScrollView, ActivityIndicator } from 'react-native';
import {  connect } from 'react-redux';
import Axios from 'axios';

import HourlyForecastBar from './HourlyForecastBar';
import Weather from './Weather';

import { fetchDailyForecast, fetchHourlyForecast } from '../store/OpenWeather/action';

const convertTime = (oriDate) => {
    let fullDate = new Date(oriDate);
    let hour = String(fullDate.getHours());
    if(hour.length === 1) {
        hour = '0' + hour;
    }
    let minute = String(fullDate.getMinutes());
    if(minute.length === 1) {
        minute = '0' + minute;
    }
    let second = String(fullDate.getSeconds());
    if(second.length === 1) {
        second = '0' + second;
    }
    return `${hour}-${minute}-${second}`;
}

export class Body extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      weathers: [],
      updatedAt: null
    }
  }

  convertDate(oriDate) {
    if(oriDate !== null) {
      let fullDate = new Date(oriDate);
      let year = fullDate.getFullYear();
      let month = String(fullDate.getMonth() + 1);
      if(month.length === 1) {
          month = '0' + month;
      }
      let date = String(fullDate.getDate());
      if(date.length === 1) {
          date = '0' + date;
      }
      return `${year}-${month}-${date}`;
    } else {
      return '-';
    }

  }

  render() {
    console.log(this.props);
    if(this.props.dailyForecast.length > 0) {
      return (
        <ScrollView style={styles.container}>
          <HourlyForecastBar hourlyForecast={this.props.hourlyForecast}/>
          <ScrollView style={styles.body}>
            {this.props.dailyForecast.map( (weather, index) => {
              return (
                <Weather key={index} weather={weather}/>
              )
            })}
          </ScrollView>
        </ScrollView>
      );
    } else {
      return (
        <View style={styles.loadingText}>
          <ActivityIndicator size="large"/>
          <Text>Loading Weather Forecast ..</Text>
        </View>
      );
    }

  }

  getNewData() {
    // console.log('getting new data')
    if(this.props.locationData) {
      let maxIndex = this.props.locationData.results.length - 1;
      // console.log(this.props.locationData.results[6].geometry.location)
      this.props.fetchDailyForecast(this.props.locationData.results[maxIndex].geometry.location);
      this.props.fetchHourlyForecast(this.props.locationData.results[maxIndex].geometry.location);
    }
  }

  checkUpdatedAt() {
    // console.log('checking updated time');
    // console.log('updated at: ', this.state.updatedAt);
    let now = new Date();
    if(this.state.updatedAt !== null) {
      let lastUpdate = new Date(this.state.updatedAt);
      console.log((now.getMinutes() - lastUpdate.getMinutes())/(1000*60));
      return ((now.getMinutes() - lastUpdate.getMinutes())/(1000*60)) > 30
    } else {
      return false
    }
  }

  // updateDailyForecast() {
  //   console.log('updating daily forecast')
  //   console.log(this.checkUpdatedAt());
  //   if(this.props.locationData) {
  //     if(this.state.updatedAt === null) {
  //         this.getNewData();
  //     }
  //   }
  // }

  componentDidUpdate() {
    console.log('body is updated')
  }

  componentDidMount() {
    this.getNewData();
  }

}

const styles = {
  container: {

  },
  body: {
    // backgroundColor: '#F6F6F6',
    padding: 5,
    margin: 10
  },
  weatherContainer: {
    // backgroundColor: '#1EE494'
  },
  weatherText: {
    fontSize: 24,
    // color: '#009378'
  },
  loadingText: {
    justifyContent: 'center',
    alignItems: 'center'
  }
}

const mapStateToProps = (state) => {
  return {
    locationData: state.Geolocation.locationData,
    dailyForecast: state.OpenWeather.dailyForecast,
    hourlyForecast: state.OpenWeather.hourlyForecast
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchDailyForecast: (coord) => dispatch(fetchDailyForecast(coord)),
    fetchHourlyForecast: (coord) => dispatch(fetchHourlyForecast(coord))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Body);
