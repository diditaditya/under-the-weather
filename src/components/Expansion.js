import React from 'react';
import { View, Text, Animated, StyleSheet } from 'react-native';

class Expansion from React.Component {

  render() {
    return (
      <View>
        <Text>For Future Expansion</Text>
      </View>
    );
  }


}

export default Expansion;

//google map key AIzaSyBCnrRCfgm831tHFx-onfyoxkPO8tmUD3A
