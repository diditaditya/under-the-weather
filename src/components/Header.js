import React from 'react';
import { View, Text } from 'react-native';

const styles = {
  header: {
    width: '100%',
    padding: 5,
    alignItems: 'center',
  },
  cityName: {
    fontSize: 16,
    justifyContent: 'center',
  },
  currentTemp: {
    fontSize: 12,
  },
};

export class Header extends React.Component {
  render() {
    const maxIndex = this.props.locationData.results.length - 1;
    return (
      <View style={styles.header}>
        <Text style={styles.cityName}>
          {this.props.locationData.results[maxIndex].formatted_address}
        </Text>
      </View>
    );
  }
}

export default Header;
