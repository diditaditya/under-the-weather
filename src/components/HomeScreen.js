import React from 'react';
import { View, Text,
  ActivityIndicator,
  Image } from 'react-native';
import { connect } from 'react-redux';
import { StackNavigator } from 'react-navigation';

import { getCurrentCoordinate } from '../store/Geolocation/action';
import { fetchCurrentWeather } from '../store/OpenWeather/action';

import WeatherForecast from './Forecast';
import StatusText from './HomeScreenStatusText';
import Location from './HomeScreenLocation';
import Thumbnail from './HomeScreenThumbnail';

const dayWallpaper = require('../images/beautiful_light_blue_sky-wallpaper-480x800.jpg');
const nightWallpaper = require('../images/005e3efe1a23e55f2327eedd932c7742.jpg');

const styles = {
  mainContainer: {
    flex: 1,
    width: null,
    height: null,
  },
  locationContainer: {
    flex: 1,
    padding: 10,
  },
  statusContainer: {
    flex: 3,
    alignItems: 'center',
    justifyContent: 'center',
  },
  thumbnail: {
    width: 300,
    height: 300,
    alignItems: 'flex-start',
    justifyContent: 'flex-end',
  },
};

export class HomeScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      wallpapers: {
        day: dayWallpaper,
        night: nightWallpaper,
      },
      selectedWallpaper: '',
    };
  }

  componentWillMount() {
    this.selectWallpaper();
  }

  componentDidMount() {
    this.props.getCurrentLocation();
  }

  fetchAllCurrentWeathers() {
    const coord = {
      lat: this.props.locationData.results[0].geometry.location.lat,
      lng: this.props.locationData.results[0].geometry.location.lng,
    };
    this.props.fetchOWCurrentWeather(coord);
  }

  selectWallpaper() {
    const time = new Date();
    if (time.getHours() > 5) {
      this.setState({
        selectedWallpaper: this.state.wallpapers.day,
      });
    } else if (time.getHours() >= 18 || time.getHours() >= 0) {
      this.setState({
        selectedWallpaper: this.state.wallpapers.night,
      });
    }
  }

  render() {
    if (this.props.locationData.results) {
      if (!this.props.OWCurrentWeather) {
        this.fetchAllCurrentWeathers();
        return (
          <Image source={this.state.selectedWallpaper} style={styles.mainContainer}>
            <View style={styles.locationContainer}>
              <Location locationData={this.props.locationData} />
            </View>
            <View style={styles.statusContainer}>
              <ActivityIndicator size="large" />
              <StatusText
                locationData={this.props.locationData}
                OWCurrentWeather={this.props.OWCurrentWeather}
              />
            </View>
          </Image>
        );
      }

      return (
        <Image source={this.state.selectedWallpaper} style={styles.mainContainer}>
          <View style={styles.locationContainer}>
            <Location locationData={this.props.locationData} />
          </View>
          <View style={styles.statusContainer}>
            <View style={styles.thumbnail}>
              <Thumbnail
                currentWeather={this.props.OWCurrentWeather}
                navigation={this.props.navigation}
              />
            </View>
          </View>
          <View style={{ alignItems: 'center', margin: 10 }}>
            <Text style={{ fontSize: 10 }}>
              Powered by <Text style={{ fontWeight: 'bold' }}>OpenWeatherMap.org</Text>
            </Text>
          </View>
        </Image>
      );
    }

    return (
      <Image source={this.state.selectedWallpaper} style={styles.mainContainer}>
        <View style={styles.statusContainer}>
          <ActivityIndicator size="large" />
          <StatusText
            locationData={this.props.locationData}
            OWCurrentWeather={this.props.OWCurrentWeather}
          />
        </View>
      </Image>
    );
  }
}

function mapStateToProps(state) {
  return {
    locationData: state.Geolocation.locationData,
    OWCurrentWeather: state.OpenWeather.currentWeather,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getCurrentLocation: () => dispatch(getCurrentCoordinate()),
    fetchOWCurrentWeather: coord => dispatch(fetchCurrentWeather(coord)),
  };
}

const Home = connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

// export default test;

const HomeNav = StackNavigator({
  HomeScreen: {
    screen: Home,
  },
  Forecast: {
    screen: WeatherForecast,
  },
},
  {
    headerMode: 'none',
  });

export default HomeNav;
