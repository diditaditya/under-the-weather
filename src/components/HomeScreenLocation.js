import React from 'react';
import { View, Text } from 'react-native';

const styles = {
  location: {
    fontSize: 30,
  },
};

export const HomeScreenLocation = (props) => {
  const maxIndex = props.locationData.results.length - 1;
  return (
    <View>
      <Text style={styles.location}>{props.locationData.results[maxIndex].formatted_address}</Text>
      <Text>Lat: {props.locationData.results[0].geometry.location.lat}</Text>
      <Text>Long: {props.locationData.results[0].geometry.location.lng}</Text>
    </View>
  );
};

export default HomeScreenLocation;
