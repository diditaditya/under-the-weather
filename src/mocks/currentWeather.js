export const OWCurrentWeather = {
  "coord": {
    "lon": 106.87,
    "lat": -6.18
  },
  "weather": [
    {
      "id": 802,
      "main": "Clouds",
      "description": "scattered clouds",
      "icon": "03d"
    }
  ],
  "base": "stations",
  "main": {
    "temp": 30,
    "pressure": 1010,
    "humidity": 70,
    "temp_min": 30,
    "temp_max": 30
  },
  "visibility": 8000,
  "wind": {
    "speed": 2.6
  },
  "clouds": {
    "all": 40
  },
  "dt": 1496379600,
  "sys": {
    "type": 1,
    "id": 8043,
    "message": 0.0132,
    "country": "ID",
    "sunrise": 1496357821,
    "sunset": 1496400242
  },
  "id": 1642907,
  "name": "Daerah Khusus Ibukota Jakarta",
  "cod": 200
}
