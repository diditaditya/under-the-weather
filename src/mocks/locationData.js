export const locationData = {
  "results": [
    {
      "address_components": [
        {
          "long_name": "16",
          "short_name": "16",
          "types": ["street_number"]
        }, {
          "long_name": "Jalan Cempaka Putih Tengah XV",
          "short_name": "Jl. Cempaka Putih Tengah XV",
          "types": ["route"]
        }, {
          "long_name": "",
          "short_name": "",
          "types": ["political"]
        }, {
          "long_name": "",
          "short_name": "",
          "types": ["political"]
        }, {
          "long_name": "Cempaka Putih Timur",
          "short_name": "Cemp. Putih Tim.",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cemp. Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Kota Jakarta Pusat",
          "short_name": "Kota Jakarta Pusat",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Daerah Khusus Ibukota Jakarta",
          "short_name": "Daerah Khusus Ibukota Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }, {
          "long_name": "10510",
          "short_name": "10510",
          "types": ["postal_code"]
        }
      ],
      "formatted_address": "Jl. Cempaka Putih Tengah XV No.16, Cemp. Putih Tim., Cemp. Putih, Kota Jakarta Pusat, Daerah Khusus Ibukota Jakarta 10510, Indonesia",
      "geometry": {
        "location": {
          "lat": -6.175078,
          "lng": 106.865087
        },
        "location_type": "ROOFTOP",
        "viewport": {
          "northeast": {
            "lat": -6.173729019708498,
            "lng": 106.8664359802915
          },
          "southwest": {
            "lat": -6.176426980291502,
            "lng": 106.8637380197085
          }
        }
      },
      "place_id": "ChIJ1dUaAAD1aS4RA4a_-bFagCE",
      "types": ["street_address"]
    }, {
      "address_components": [
        {
          "long_name": "Central Jakarta",
          "short_name": "Central Jakarta",
          "types": ["locality", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Central Jakarta, Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.8810121
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.7919172
          }
        },
        "location": {
          "lat": -6.1864864,
          "lng": 106.8340911
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.8810121
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.7919201
          }
        }
      },
      "place_id": "ChIJB0vJuDb0aS4R9oJ8iznVpm4",
      "types": ["locality", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["colloquial_area", "locality", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.0886599,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.3708331,
            "lng": 106.686211
          }
        },
        "location": {
          "lat": -6.2087634,
          "lng": 106.845599
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.0886599,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.3708331,
            "lng": 106.686211
          }
        }
      },
      "place_id": "ChIJnUvjRenzaS4RoobX2g-_cVM",
      "types": ["colloquial_area", "locality", "political"]
    }, {
      "address_components": [
        {
          "long_name": "2",
          "short_name": "2",
          "types": ["political"]
        }, {
          "long_name": "",
          "short_name": "",
          "types": ["political"]
        }, {
          "long_name": "Cempaka Putih Timur",
          "short_name": "Cemp. Putih Tim.",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cemp. Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Kota Jakarta Pusat",
          "short_name": "Kota Jakarta Pusat",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Daerah Khusus Ibukota Jakarta",
          "short_name": "Daerah Khusus Ibukota Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "RT.2, Cemp. Putih Tim., Cemp. Putih, Kota Jakarta Pusat, Daerah Khusus Ibukota Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.173481,
            "lng": 106.8679419
          },
          "southwest": {
            "lat": -6.176254000000001,
            "lng": 106.864878
          }
        },
        "location": {
          "lat": -6.1754171,
          "lng": 106.8654083
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.173481,
            "lng": 106.8679419
          },
          "southwest": {
            "lat": -6.176254000000001,
            "lng": 106.864878
          }
        }
      },
      "place_id": "ChIJ2YMNHwD1aS4R96wzCawb0yE",
      "types": ["political"]
    }, {
      "address_components": [
        {
          "long_name": "8",
          "short_name": "8",
          "types": ["political"]
        }, {
          "long_name": "Cempaka Putih Timur",
          "short_name": "Cemp. Putih Tim.",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cemp. Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Kota Jakarta Pusat",
          "short_name": "Kota Jakarta Pusat",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Daerah Khusus Ibukota Jakarta",
          "short_name": "Daerah Khusus Ibukota Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "RW.8, Cemp. Putih Tim., Cemp. Putih, Kota Jakarta Pusat, Daerah Khusus Ibukota Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1729039,
            "lng": 106.8721418
          },
          "southwest": {
            "lat": -6.176314,
            "lng": 106.864153
          }
        },
        "location": {
          "lat": -6.1755806,
          "lng": 106.8702028
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1729039,
            "lng": 106.8721418
          },
          "southwest": {
            "lat": -6.176314,
            "lng": 106.864153
          }
        }
      },
      "place_id": "ChIJDbEM0wH1aS4R2SnRob7aLs0",
      "types": ["political"]
    }, {
      "address_components": [
        {
          "long_name": "10510",
          "short_name": "10510",
          "types": ["postal_code"]
        }, {
          "long_name": "East Cempaka Putih",
          "short_name": "East Cempaka Putih",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cempaka Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "East Cempaka Putih, Cempaka Putih, Central Jakarta City, Jakarta 10510, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1663198,
            "lng": 106.8790512
          },
          "southwest": {
            "lat": -6.188108400000001,
            "lng": 106.8644485
          }
        },
        "location": {
          "lat": -6.1761468,
          "lng": 106.8724156
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1663198,
            "lng": 106.8790512
          },
          "southwest": {
            "lat": -6.188108400000001,
            "lng": 106.8644485
          }
        }
      },
      "place_id": "ChIJm-SGA-P0aS4RsGQDTevFABw",
      "types": ["postal_code"]
    }, {
      "address_components": [
        {
          "long_name": "East Cempaka Putih",
          "short_name": "East Cempaka Putih",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cempaka Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "East Cempaka Putih, Cempaka Putih, Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.188168000000001,
            "lng": 106.8642099
          }
        },
        "location": {
          "lat": -6.1761468,
          "lng": 106.8724156
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.188168000000001,
            "lng": 106.8642099
          }
        }
      },
      "place_id": "ChIJM3dU0v30aS4Rx3RI4iVSPdI",
      "types": ["administrative_area_level_4", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Cempaka Putih",
          "short_name": "Cempaka Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Cempaka Putih, Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.196158,
            "lng": 106.85534
          }
        },
        "location": {
          "lat": -6.182670799999999,
          "lng": 106.8679899
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.196158,
            "lng": 106.85534
          }
        }
      },
      "place_id": "ChIJc-r_wvj0aS4Ru4tkAxqdPIs",
      "types": ["administrative_area_level_3", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.881432
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.791799
          }
        },
        "location": {
          "lat": -6.1864674,
          "lng": 106.8296372
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.881432
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.791799
          }
        }
      },
      "place_id": "ChIJB0vJuDb0aS4RlHPzGrFt85A",
      "types": ["administrative_area_level_2", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -5.1843219,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.3708331,
            "lng": 106.3831259
          }
        },
        "location": {
          "lat": -6.17511,
          "lng": 106.8650395
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.0886599,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.370829899999999,
            "lng": 106.686211
          }
        }
      },
      "place_id": "ChIJnUvjRenzaS4RILjULejFAAE",
      "types": ["administrative_area_level_1", "political"]
    }
  ],
  "status": "OK"
}
