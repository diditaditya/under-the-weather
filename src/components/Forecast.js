import React from 'react';
import { Image } from 'react-native';
import { connect } from 'react-redux';

import NavBar from './NavBar';
import Header from './Header';
import Body from './Body';

const wallpaper = require('../images/beautiful_light_blue_sky-wallpaper-480x800.jpg');

const styles = {
  container: {
    flex: 1,
    width: null,
    height: null,
  },
};

export class Forecast extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      wallpaper,
    };
  }

  componentWillMount() {
    if (this.props.wallpaper) {
      const image = require(this.props.wallpaper);
      this.setState({
        wallpaper: image,
      });
    }
  }

  render() {
    // console.log(this.props);
    return (
      <Image source={this.state.wallpaper} style={styles.container}>
        <NavBar navigation={this.props.navigation} />
        <Header
          locationData={this.props.locationData}
          OWCurrentWeather={this.props.OWCurrentWeather}
        />
        <Body />
      </Image>
    );
  }

}

function mapStateToProps(state) {
  return ({
    wallpaper: state.Geolocation.selectedWallpaper,
    locationData: state.Geolocation.locationData,
    OWCurrentWeather: state.OpenWeather.currentWeather,
  });
}

export default connect(mapStateToProps, null)(Forecast);
