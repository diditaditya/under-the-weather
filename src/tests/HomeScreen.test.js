import React from 'react';
import renderer from 'react-test-renderer';

import {HomeScreen} from '../components/HomeScreen';

import { locationData } from '../mocks/locationData';
import { OWCurrentWeather } from '../mocks/currentWeather';

const getCurrentLocation = () => {}
const fetchOWCurrentWeather = () => {}

describe('HomeScreen', () => {

  it('should render loading location data because of empty location data', () => {
    const Home = renderer.create(<HomeScreen locationData={''} OWCurrentWeather={''} getCurrentLocation={getCurrentLocation}/>);
      expect(Home).toMatchSnapshot();
  });

  it('should render loading current weather data because of empty weather data', () => {
    const Home = renderer.create(<HomeScreen locationData={locationData} OWCurrentWeather={''} getCurrentLocation={getCurrentLocation} fetchOWCurrentWeather={fetchOWCurrentWeather}/>);
      expect(Home).toMatchSnapshot();
  });

  it('should render with current weather and location data', () => {
    const Home = renderer.create(<HomeScreen locationData={locationData} OWCurrentWeather={OWCurrentWeather} getCurrentLocation={getCurrentLocation}/>);
    expect(Home).toMatchSnapshot();
  });

});
