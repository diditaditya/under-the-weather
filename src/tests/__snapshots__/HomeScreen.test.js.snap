// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render loading current weather data because of empty weather data 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        Jakarta, Indonesia
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Lat: 
        -6.1761468
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Long: 
        106.8724156
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "center",
            "fontSize": 16,
            "justifyContent": "center",
          }
        }
      >
        Connecting to Weather Servers..
      </Text>
    </View>
  </View>
</Image>
`;

exports[`HomeScreen should render loading location data because of empty location data 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "center",
            "fontSize": 16,
            "justifyContent": "center",
          }
        }
      >
        Loading Location Data..
      </Text>
    </View>
  </View>
</Image>
`;

exports[`HomeScreen should render with current weather and location data 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        Jakarta, Indonesia
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Lat: 
        -6.1761468
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Long: 
        106.8724156
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "height": 300,
          "justifyContent": "flex-end",
          "width": 300,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://openweathermap.org/img/w/undefined.png",
                }
              }
              style={
                Object {
                  "height": 100,
                  "width": 100,
                }
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 50,
                    "margin": 10,
                    "opacity": 0.6,
                    "width": 50,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 150,
                  "opacity": 2,
                }
              }
            >
              30
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 50,
                  "marginTop": 25,
                }
              }
            >
              o
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 25,
                  "marginRight": 10,
                }
              }
            >
              min: 
              30
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 25,
                  "marginRight": 10,
                }
              }
            >
              max: 
              30
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</Image>
`;
